persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/StargateSystemKeyspace.java

```diff
@@ -89,4 +98,20 @@ public static TableMetadata virtualFromLegacy(TableMetadata legacy, String table
     legacy.regularColumns().forEach(cm -> builder.addRegularColumn(cm.name, cm.type.freeze()));
     return builder.build();
   }
+
+  public static Set<String> generateRandomTokens(InetAddress inetAddress, int numTokens) {
+    Random random = new Random(getSeed(inetAddress));
+    IPartitioner partitioner = DatabaseDescriptor.getPartitioner();
+    TokenFactory tokenFactory = partitioner.getTokenFactory();
+    Set<String> tokens = new HashSet<>(numTokens);
+    while (tokens.size() < numTokens) {
+      tokens.add(tokenFactory.toString(partitioner.getRandomToken(random)));
+    }
+    return tokens;
+  }
+
+  private static long getSeed(InetAddress inetAddress) {
```

Comment:
Just wondering why do we sometimes use just the address and sometimes address and port?

Reply:
C* 4.0 uses address and port as the full address identity and  C* 3.11/DSE don't.

Reply:
TIL, thx!